src/benchmark/measurement.ts

```diff
@@ -1,62 +1,92 @@
-import Transpiler from '../transpiler';
+import {Timing, Transpiler} from '../transpiler';
 
 // noinspection TsLint
 type Algorithm = {
     // noinspection TsLint
     func: Function[],
     arguments: any[],
-    expectedResult: any
+    expectedResult: any,
 };
 
 class Measurement {
 
+    public static measure(algorithm: Algorithm,
+                          warmupRounds: number,
+                          measureRounds: number): [Timing[], Timing[]] {
+        const func = algorithm.func;
+        const args = algorithm.arguments;
+        const expectedResult = algorithm.expectedResult;
+
+        Measurement.executeJS(func[0], args, expectedResult, warmupRounds);
+        const jsTimes = Measurement.executeJS(func[0], args, expectedResult, measureRounds);
+
+        const wasmArgs = args.slice();
+        wasmArgs.unshift(func[0].name);
+
+        const wasmAlgorithm = Transpiler.transpile(func.join(''));
+        Measurement.executeWasm(wasmAlgorithm, wasmArgs, expectedResult, warmupRounds);
+        const wasmTimes = Measurement.executeWasm(wasmAlgorithm, wasmArgs, expectedResult, measureRounds);
+
+        return [jsTimes, wasmTimes];
+    }
+
     // noinspection TsLint
-    private static executeAlgorithm(algorithm: Function,
-                                    args: any[],
-                                    expectedResult: number,
-                                    rounds: number): number[] {
-        const times: number[] = [];
+    private static executeJS(algorithm: Function,
+                             args: any[],
+                             expectedResult: any,
+                             rounds: number): Timing[] {
+        const times: Timing[] = [];
 
         for (let i = 0; i < rounds; i++) {
-            const startTime: number = Date.now();
+            const start = performance.now();
             const result = algorithm(...args);
```

Comment:
We need to work with a deep copy of the args array. Otherwise object parameters do not always have the initial values.

Reply:
We don't support objects yet so this shouldn't be a problem at the moment

Reply:
Well we support arrays and this problem can occur as soon as we have the export functionality implementet. We can postpone this until we need it. I just brought it up so it doesn't get forgotten.

Reply:
Good, we will implement this when we work on the export functionality. Most of our benchmark cases will fill the arrays in the wasm code, so it shouldn't be a problem.